lib/client-assets.php

```diff
@@ -187,12 +187,13 @@ function gutenberg_register_packages_scripts() {
 		// For example, `…/build/a11y/index.js` becomes `wp-a11y`.
 		$handle = 'wp-' . basename( dirname( $path ) );
 
-		// Replace `.js` extension with `.deps.json` to find the generated dependencies file.
-		$dependencies_file = substr( $path, 0, -3 ) . '.deps.json';
-
-		$dependencies = is_readable( $dependencies_file )
-			? json_decode( file_get_contents( $dependencies_file ) )
-			: array();
+		// Replace `.js` extension with `.asset.php` to find the generated dependencies file.
+		$asset_file   = substr( $path, 0, -3 ) . '.asset.php';
+		$asset        = file_exists( $asset_file )
```

Comment:
Any particular reason to switch away from `is_readable`?

Actually, as I was thinking about this, I like `file_exists` better for a couple of reasons. For instance, it's better to let PHP raise a warning if for unexpected reasons an asset file exists but isn't readable.

Reply:
I saw that `file_exists` is used in many places in WordPress core and I thought we update it since we are now using native PHP functions to load file.